File: Java/ApplicationMetrics/src/net/alastairwyse/applicationmetrics/NullMetricLogger.java
/*
 * Copyright 2014 Alastair Wyse (http://www.oraclepermissiongenerator.net/methodinvocationremoting/)
 * 
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * 
 *     http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package net.alastairwyse.applicationmetrics;

/**
 * Implementation of the IMetricLogger interface which does not perform any metric logging.
 * An instance of this class can be used as the default IMetricLogger implementation inside a client class, to prevent occurrences of NullPointerException.
 * @author Alastair Wyse
 */
public class NullMetricLogger implements IMetricLogger {

    /**
     * Initialises a new instance of the NullMetricLogger class.
     */
    public NullMetricLogger() {
        
    }
    
    @Override
    public void Increment(CountMetric countMetric) {

    }

    @Override
    public void Add(AmountMetric amountMetric) {

    }

    @Override
    public void Set(StatusMetric statusMetric) {

    }

    @Override
    public void Begin(IntervalMetric intervalMetric) {

    }

    @Override
    public void End(IntervalMetric intervalMetric) {

    }
}

File: C#/ApplicationMetrics/NullMetricLogger.cs
/*
 * Copyright 2015 Alastair Wyse (http://www.oraclepermissiongenerator.net/methodinvocationremoting/)
 * 
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * 
 *     http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

using System;
using System.Collections.Generic;
using System.Text;

namespace ApplicationMetrics
{
    //******************************************************************************
    //
    // Class: NullMetricLogger
    //
    //******************************************************************************
    /// <summary>
    /// Implementation of the IMetricLogger interface which does not perform any metric logging.
    /// </summary>
    /// <remarks>An instance of this class can be used as the default IMetricLogger implementation inside a client class, to prevent occurrences of the 'Object reference not set to an instance of an object' error.</remarks>
    public class NullMetricLogger : IMetricLogger
    {
        //------------------------------------------------------------------------------
        //
        // Method: NullMetricLogger (constructor)
        //
        //------------------------------------------------------------------------------
        /// <summary>
        /// Initialises a new instance of the ApplicationMetrics.NullMetricLogger class.
        /// </summary>
        public NullMetricLogger()
        {
        }

        /// <include file='InterfaceDocumentationComments.xml' path='doc/members/member[@name="M:ApplicationMetrics.IMetricLogger.Increment(ApplicationMetrics.CountMetric)"]/*'/>
        public void Increment(CountMetric countMetric)
        {
        }

        /// <include file='InterfaceDocumentationComments.xml' path='doc/members/member[@name="M:ApplicationMetrics.IMetricLogger.Add(ApplicationMetrics.AmountMetric)"]/*'/>
        public void Add(AmountMetric amountMetric)
        {
        }

        /// <include file='InterfaceDocumentationComments.xml' path='doc/members/member[@name="M:ApplicationMetrics.IMetricLogger.Set(ApplicationMetrics.StatusMetric)"]/*'/>
        public void Set(StatusMetric statusMetric)
        {
        }

        /// <include file='InterfaceDocumentationComments.xml' path='doc/members/member[@name="M:ApplicationMetrics.IMetricLogger.Begin(ApplicationMetrics.IntervalMetric)"]/*'/>
        public void Begin(IntervalMetric intervalMetric)
        {
        }

        /// <include file='InterfaceDocumentationComments.xml' path='doc/members/member[@name="M:ApplicationMetrics.IMetricLogger.End(ApplicationMetrics.IntervalMetric)"]/*'/>
        public void End(IntervalMetric intervalMetric)
        {
        }

        /// <include file='InterfaceDocumentationComments.xml' path='doc/members/member[@name="M:ApplicationMetrics.IMetricLogger.CancelBegin(ApplicationMetrics.IntervalMetric)"]/*'/>
        public void CancelBegin(IntervalMetric intervalMetric)
        {
        }
    }
}
